aptos-move/parallel-executor/src/proptest_types/types.rs

```diff
@@ -235,16 +156,20 @@ where
     ) -> ExecutionStatus<Self::Output, Self::Error> {
         match txn {
             Transaction::Write {
+                incarnation,
                 reads,
-                actual_writes,
-                skipped_writes: _,
+                writes,
             } => {
+                let idx = incarnation.fetch_add(1, Ordering::SeqCst);
+                let write_idx = idx % writes.len();
```

Comment:
is this the dynamic part? looks like magic to me

Reply:
added comments.